scalalib/src/mill/javalib/android/AndroidAppBundle.scala

```diff
@@ -0,0 +1,117 @@
+package mill.javalib.android
+
+import mill._
+import mill.scalalib._
+import mill.api.PathRef
+import mill.javalib.android.AndroidAppModule
+
+/**
+ * Defines an Android app bundle module with tasks for resource compilation,
+ * bundling using bundle tool.
+ */
+@mill.api.experimental
+trait AndroidAppBundle extends AndroidAppModule with JavaModule {
+
+  override def linkOptions: T[Seq[String]] = Task {
+    super.linkOptions() ++ Seq[String](
+      "--proto-format"
+    ) // protoformat for resources linking(required for bundle)
+  }
+
+  /**
+   * Packages compiled resources and dex files into a zip for app bundle generation.
+   *
+   * @return PathRef to the created bundle zip.
+   */
+  def androidBundleZip: T[PathRef] = Task {
+    val dexFile = androidDex().path
+    val resFile = androidResources().path / "res.apk"
+    val baseDir = Task.dest / "base"
+
+    // Copy dex and resource files into the bundle directory
+    os.copy(dexFile, Task.dest / dexFile.last)
+    os.unzip(resFile, Task.dest)
```

Comment:
This `unzip` into `Task.dest` is a bit sloppy. We don't know what `resFile` contains, and so if it ends up containing files or files named `manifest` or `base` everything might break. You should unzip it into a subfolder of `Task.dest` so we can be guaranteed that it will be an empty folder and there will be no collisions